proto/tendermint/store/types.proto

```diff
@@ -8,8 +8,12 @@ message BlockStoreState {
   int64 height = 2;
 }
 
-// TxInfo describes the location of a tx inside a committed block.
+// TxInfo describes the location of a tx inside a committed block
+// as well as the result of executing the transaction.
 message TxInfo {
   int64 height    = 1;
-  int64 index     = 2;
+  uint32 index    = 2;
```

Comment:
why does the index have type have to change from `int64` => `uint32`?

Reply:
Index can't be negative. Making it unsigned embeds that constraint. 4 bytes should be beyond sufficient to cover the amount of transactions in a block (as opposed to 8 `uint64`)